hack/gomod.sh

```diff
@@ -5,44 +5,63 @@
 #
 # NOTE: This won't work unless the build environment has internet access
 
-set -ux
+set -eux
 
-IS_CONTAINER=${IS_CONTAINER:-false}
+IS_CONTAINER="${IS_CONTAINER:-false}"
 CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+MOUNTDIR="/workdir"
+WORKDIR="/tmp${MOUNTDIR}"
 
 if [ "${IS_CONTAINER}" != "false" ]; then
-  export XDG_CACHE_HOME="/tmp/.cache"
+    export XDG_CACHE_HOME="/tmp/.cache"
 
-  STATUS=$(git status --porcelain)
-  if [ -n "$STATUS" ]; then
-      echo "Dirty tree: refusing to continue out of caution"
-      exit 1
-  fi
+    # locally we need to make a copy for writing, while in prow we don't
+    if [ "${COPY_REPOSITORY:-}" = "TRUE" ]; then
+        cp -a "${MOUNTDIR}"/ "${WORKDIR}"/
+        # newer git versions have safe directory mechanism we need to disable
+        git config --global --add safe.directory "${WORKDIR}"
+        # move workdir to the repository copy
+        cd "${WORKDIR}" || { echo "error: can't enter ${WORKDIR}"; exit 1; }
+    fi
 
-  go mod tidy
-  rc=$?
+    STATUS="$(git status --porcelain)"
+    if [ -n "${STATUS}" ]; then
+        echo "Dirty tree: refusing to continue out of caution"
+        exit 1
+    fi
 
-  if [ $rc -ne 0 ]; then
-      echo "'go mod tidy' failed"
-      exit 1;
-  fi
+    # sh doesn't allow ** recursive glob
+    for mod_file in *.mod */*.mod */*/*.mod; do
+        # if glob doesn't match, it comes back as */*.mod, which we must skip
+        if [ ! -r "${mod_file}" ]; then
+            continue
+        fi
+        mod_dir="$(dirname "${mod_file}")"
+        cd "${mod_dir}"
+        echo "${mod_dir}:"
 
-  STATUS=$(git status --porcelain go.mod go.sum)
-  if [ -n "$STATUS" ]; then
-      echo "go.mod and go.sum changed"
-      echo "Please run 'go mod tidy' and commit the changes to go.mod & go.sum."
-      echo "Abort"
-      exit 1
-  fi
+        if ! go mod tidy; then
+            echo "'go mod tidy' failed"
+            exit 1
+        fi
 
-  exit 0;
+        STATUS="$(git status --porcelain go.mod go.sum)"
+        if [ -n "${STATUS}" ]; then
+            echo "go.mod and/or go.sum changed"
```

Comment:
I see that we echo the dir above, but maybe we could add it here to make it even more obvious?
```suggestion
            echo "go.mod and/or go.sum changed in ${mod_dir}"
```

Reply:
I made it output the `git status` output so it prints out all of the changed files.